src/com/xilinx/rapidwright/rwroute/RWRouteConfig.java

```diff
@@ -373,6 +375,16 @@ public void setWirelengthWeight(float wirelengthWeight) {
         if (wirelengthWeight < 0 || wirelengthWeight > 1)
             throw new IllegalArgumentException("ERROR: wirelength-driven weighting factor wirelengthWeight should be within [0, 1].");
         this.wirelengthWeight = wirelengthWeight;
+
+        // Assume that the minimum unit we want to observe is 1/8th of the wirelengthWeight
+        // (since during RWRoute.evaluateCostAndPush(), distanceToSink is scaled by wirelengthWeight)
+        // compute the largest floating-point value that results in this Units-in-the-Last-Place value.
+        final float maxUlp = wirelengthWeight / 8;
```

Comment:
How do we get 1/8 as the minimum unit?  Does this still make sense if someone chooses a different initial value for `wirelengthWeight` like `0.5`?

Reply:
1/8 is just a power-of-2 heuristic value. However, if someone chooses a wirelength weight of 0.5 (down from the default of 0.8), then `maxUlp` will be reduced accordingly, and the `maxPresentCongestionCost` will continue to be scaled down until its ULP is below this value.